Hypothesis Testing Assignment Solution. 

In [1]:
import pandas as pd
import scipy 
import numpy as np
from scipy import stats


Ques1 F&B Manager Cutlets 

In [2]:
df = pd.read_csv("/content/Cutlets.csv")
df

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [3]:
df.columns

Index(['Unit A', 'Unit B'], dtype='object')

In [4]:
#Calculating t Staistics Value & p Value
Value = stats.ttest_ind(df["Unit A"] , df["Unit B"]);
Value 

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

We got the P value is 0.472 which is greater than the alpha value (0.05), therefore we can accept the Null hypothesis. It means that there is no significant difference in the diameters of the cutlet between the two units. 

Ques 2 Hospital LabTAT.mtw

In [5]:
import scipy.stats as stats
df=pd.read_csv("/content/LabTAT.csv")
df.columns

Index(['Laboratory 1', 'Laboratory 2', 'Laboratory 3', 'Laboratory 4'], dtype='object')

In [6]:
#Now Doing ANOVA Test
stats.f_oneway(df.iloc[:,0], df.iloc[:,1], df.iloc[:,2], df.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Here p value is equivalent to 0 i.e; P value is less than alpha value i.e; 0.05, it means reject null hypothesis.

> Indented block



Ques 3 Buyers Ratio.mtw

In [7]:
buyer=pd.read_csv("/content/BuyerRatio.csv")
buyer

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

Now Doing ANOVA Test

In [8]:
stats.f_oneway(buyer.iloc[:,1], buyer.iloc[:,2], buyer.iloc[:,3], buyer.iloc[:,4])

F_onewayResult(statistic=0.3068565327605033, pvalue=0.8204060154887464)

 Ques 4 Customer Order Form.mtw

In [9]:
from scipy.stats import chi2_contingency

In [10]:
data = pd.read_csv("/content/Costomer+OrderForm.csv")
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [11]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [12]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [13]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [14]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [15]:
# Make a Contengency Table

obs=np.array([[271,267,269,280], [29,33,31,20]])
obs 

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [16]:
#Chi2 Contengency Independence Test

chi2_contingency(obs)   # Output is (Chi2 calculated value or stats value, p_value, dof, Expected obsedrvations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

We get the Chi2 value is 3.85 and p-value as 0.277 which is greater than the alpha value 0.05i.e; 5% significance value. 

Hence we accept the alterbnate hypothesis.